nestle.py

```diff
@@ -655,19 +732,23 @@ def set_options(self, options):
         self.enlarge = options.get('enlarge', 1.2)
 
     def update(self, pointvol):
+        self.empty_queue()
         self.ells = bounding_ellipsoids(self.points, pointvol=pointvol)
         for ell in self.ells:
             ell.scale_to_vol(ell.vol * self.enlarge)
+        self.fill_queue()
 
-    def new_point(self, loglstar):
+    def propose_point(self):
+        while True:
+            u = sample_ellipsoids(self.ells, rstate=self.rstate)
+            if np.all(u > 0.) and np.all (u < 1.):
+                break
+        return u
+
+    def new_point(self, loglstar, should_update):
         ncall = 0
         while True:
-            while True:
-                u = sample_ellipsoids(self.ells, rstate=self.rstate)
-                if np.all(u > 0.) and np.all(u < 1.):
-                    break
-            v = self.prior_transform(u)
-            logl = self.loglikelihood(v)
+            u, v, logl = self.get_point_value(should_update)
```

Comment:
Good catch. So what's happening now then is the code gets to this point, calls `get_point_value` a bunch of times, and when it runs out fills up the queue again. This doesn't give the behavior I want. I think you can just solve this by putting an addition condition in the while loop -- I'll try that now.